src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -1773,6 +1779,95 @@ namespace Aint {
+                "Second Argument to anint must be of integer type",
+                x.base.base.loc, diagnostics);
+        }
+    }
+
+    static ASR::expr_t *eval_Anint(Allocator &al, const Location &loc,
+            ASR::ttype_t* arg_type, Vec<ASR::expr_t*> &args) {
+        double rv = ASR::down_cast<ASR::RealConstant_t>(expr_value(args[0]))->m_r;
+        return f(std::round(rv), arg_type);
+    }
+
+    static inline ASR::asr_t* create_Anint(
+            Allocator& al, const Location& loc, Vec<ASR::expr_t*>& args,
+            const std::function<void (const std::string &, const Location &)> err) {
+        ASR::ttype_t* return_type = expr_type(args[0]);
+        if (!(args.size() == 1 || args.size() == 2)) {
+            err("Intrinsic `anint` function accepts exactly 1 or 2 arguments", loc);
+        } else if (!ASRUtils::is_real(*return_type)) {
+            err("Argument of the `anint` function must be Real", args[0]->base.loc);
+        }
+        Vec<ASR::expr_t *> m_args; m_args.reserve(al, 1);
+        m_args.push_back(al, args[0]);
+        if ( args[1] ) {
+            int kind = -1;
+            if (!ASR::is_a<ASR::Integer_t>(*expr_type(args[1])) ||
+                    !extract_value(args[1], kind)) {
+                err("`kind` argument of the `anint` function must be a "
+                    "scalar Integer constant", args[1]->base.loc);
+            }
+            return_type = TYPE(ASR::make_Real_t(al, return_type->base.loc, kind));
+        }
+        ASR::expr_t *m_value = nullptr;
+        if (all_args_evaluated(m_args)) {
+            m_value = eval_Anint(al, loc, return_type, m_args);
+        }
+        return ASR::make_IntrinsicScalarFunction_t(al, loc,
+            static_cast<int64_t>(IntrinsicScalarFunctions::Anint),
+            m_args.p, m_args.n, 0, return_type, m_value);
+    }
+
+    static inline ASR::expr_t* instantiate_Anint(Allocator &al, const Location &loc,
+            SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types, ASR::ttype_t *return_type,
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        std::string func_name = "_lcompilers_anint_" + type_to_str_python(arg_types[0]);
+        std::string fn_name = scope->get_unique_name(func_name);
+        SymbolTable *fn_symtab = al.make_new<SymbolTable>(scope);
+        Vec<ASR::expr_t*> args;
+        args.reserve(al, new_args.size());
+        ASRBuilder b(al, loc);
+        Vec<ASR::stmt_t*> body; body.reserve(al, 1);
+        SetChar dep; dep.reserve(al, 1);
+        if (scope->get_symbol(fn_name)) {
+            ASR::symbol_t *s = scope->get_symbol(fn_name);
+            ASR::Function_t *f = ASR::down_cast<ASR::Function_t>(s);
+            return b.Call(s, new_args, expr_type(f->m_return_var), nullptr);
+        }
+        fill_func_arg("a", arg_types[0]);
+        auto result = declare(fn_name, return_type, ReturnVar);
+
+        ASR::expr_t* adjusted_value;
+        if ( fLtE(args[0], f(0.0, arg_types[0])) ) {
```

Comment:
This is not how you do, here you are creating `if` in c++ code and not in ASR, to get it done you have to do: 
```suggestion
       ASR::expr_t* test = make_Compare(make_IntegerCompare_t, args[0], GtE, zero);
```
Refer: `instantiate_Abs` and try to understand difference. To see the ASR you generate with the current code do

```
% lfortran a.f90 --pass=intrinsic_functions --show-asr
```